core/txpool/celo_validation.go

```diff
@@ -0,0 +1,109 @@
+package txpool
+
+import (
+	"errors"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+var NonWhitelistedFeeCurrencyError = errors.New("Fee currency given is not whitelisted at current block")
+
+// FeeCurrencyValidator validates currency whitelisted status at the specified
+// block number.
+type FeeCurrencyValidator interface {
+	IsWhitelisted(st *state.StateDB, feeCurrency *common.Address) bool
+	// Balance returns the feeCurrency balance of the address specified, in the given state.
+	// If feeCurrency is nil, the native currency balance has to be returned.
+	Balance(st *state.StateDB, address common.Address, feeCurrency *common.Address) *big.Int
+}
+
+func NewFeeCurrencyValidator() FeeCurrencyValidator {
+	return &feeval{}
+}
+
+type feeval struct {
+}
+
+func (f *feeval) IsWhitelisted(st *state.StateDB, feeCurrency *common.Address) bool {
+	// TODO: implement proper validation for all currencies
+	return feeCurrency == nil
```

Comment:
Shouldn't this return true if we want to allow fee currency transaction to pass through?

Reply:
I can hardcode it for the moment to allow the tests to run